### PR TITLE
Task/contact filter

### PR DESCRIPTION
**Pull request by JuanGro**
#### What's this PR do?
To integrate the search in contacts component.

#### Where should the reviewer start?
With the way to find an item.

#### How should this be manually tested?
Searching an specific contact by name, last name, charge, company or email.

#### Any background context you want to provide?
No.

This template was adapted ~stolen~ from [Quickleft/Sprint.ly](https://quickleft.com/blog/pull-request-templates-make-code-review-easier/)

### CODE REVIEW

File: src/app/accounts/contacts/contacts.component.ts [hunk 1]
```diff
@@ -24,9 +24,13 @@ import { environment } from '../../../environments/environment';
...
 export class ContactsComponent implements OnInit {
   // Received from table component, it gives me the contact that the user selected to see his detail.
-  @Input() currentContact: Contact;
+  @Input('currentContact') currentContact: Contact;
+  // Received from table component, it gives me the contact that the user selected to see his detail.
+  @Input('contactsListFiltered') contactsListFiltered: Contact[];
   // Variable that saves the title to show in the template.
   public title: string;
+  // Original copy of the contacts list received from httpService.
+  public originalContactsList: Contact[];
```
Comment (marccolm) [inline src/app/accounts/contacts/contacts.component.ts]: The same, not to convinced, maybe use completeContactList and filteredContactList, etc.

File: src/app/accounts/contacts/contacts-list.component.ts [hunk 2]
```diff
@@ -53,5 +60,33 @@ export class ContactsListComponent {
...
+      if (contact.name.toLowerCase().includes(value.toLowerCase()) ||
+          contact.last_name.toLowerCase().includes(value.toLowerCase()) ||
+          contact.charge.toLowerCase().includes(value.toLowerCase()) ||
+          contact.client_complete.name.toLowerCase().includes(value.toLowerCase()) ||
+          contact.email.toLowerCase().includes(value.toLowerCase()) ||
+          contact.alternate_email.toLowerCase().includes(value.toLowerCase())) {
+          this.contactsList.push(contact);
+      }
+    }
+    this.contactsList.sort();
```
Comment (marccolm) [inline src/app/accounts/contacts/contacts-list.component.ts]: Same, I would chance contactsList to contactList

File: src/app/accounts/contacts/contacts-list.component.ts [hunk 3]
```diff
@@ -53,5 +60,33 @@ export class ContactsListComponent {
...
+    this.contactsList = [];
+    for (let contactFromList of this.originalContactsList) {
+      let contact = new Contact(contactFromList);
+      if (contact.name.toLowerCase().includes(value.toLowerCase()) ||
+          contact.last_name.toLowerCase().includes(value.toLowerCase()) ||
+          contact.charge.toLowerCase().includes(value.toLowerCase()) ||
+          contact.client_complete.name.toLowerCase().includes(value.toLowerCase()) ||
+          contact.email.toLowerCase().includes(value.toLowerCase()) ||
+          contact.alternate_email.toLowerCase().includes(value.toLowerCase())) {
+          this.contactsList.push(contact);
```
Comment (marccolm) [inline src/app/accounts/contacts/contacts-list.component.ts]: I would change some names on the components, like contactsListFiltered, I would change it to filteredContactList.

File: src/app/accounts/contacts/contacts.component.ts (hunk 1 above)
Comment (JuanGro) [inline src/app/accounts/contacts/contacts.component.ts]: Already discussed.

File: src/app/accounts/contacts/contacts-list.component.ts (hunk 3 above)
Comment (JuanGro) [inline src/app/accounts/contacts/contacts-list.component.ts]: Already discussed.

File: src/app/accounts/contacts/contacts-list.component.ts (hunk 2 above)
Comment (JuanGro) [inline src/app/accounts/contacts/contacts-list.component.ts]: Already discussed.

File: src/app/accounts/contacts/contacts.component.ts [hunk 4]
```diff
@@ -24,9 +24,13 @@ import { environment } from '../../../environments/environment';
 **/
 export class ContactsComponent implements OnInit {
   // Received from table component, it gives me the contact that the user selected to see his detail.
-  @Input() currentContact: Contact;
+  @Input('currentContact') currentContact: Contact;
+  // Received from table component, it gives me the contact that the user selected to see his detail.
```
Comment (EduardoVaca) [inline src/app/accounts/contacts/contacts.component.ts]: This comment is not related to the variable

File: src/app/accounts/contacts/contacts.component.ts [hunk 5]
```diff
@@ -24,9 +24,13 @@ import { environment } from '../../../environments/environment';
...
+  @Input('contactsListFiltered') contactsListFiltered: Contact[];
```
Comment (EduardoVaca) [inline src/app/accounts/contacts/contacts.component.ts]: Why do you need this?

File: src/app/accounts/contacts/contacts-list.component.ts [hunk 6]
```diff
@@ -25,10 +30,12 @@ export class ContactsListComponent {
   @Output() requestShowUpdateContactModal: EventEmitter<string> = new EventEmitter();
   // Sends the contact selected by the user to parent component.
   @Output() currentContact: EventEmitter<Contact> = new EventEmitter<Contact>();
+  // Sends the contactsList updated to parent component to avoid problems.
```
Comment (EduardoVaca) [inline src/app/accounts/contacts/contacts-list.component.ts]: What problems are you talking about?

Comment (JuanGro) [inline src/app/accounts/contacts/contacts-list.component.ts]: This variable is shared by parent component and the table, so we need to have the same contact list or when you use the TWDB it could cause problems (it won't push the new contact to the list).

File: src/app/accounts/contacts/contacts.component.ts (hunk 5 above)
Comment (JuanGro) [inline src/app/accounts/contacts/contacts.component.ts]: To differentiate the inputs in the selectors on html.

File: src/app/accounts/contacts/contacts.component.ts (hunk 4 above)
Comment (JuanGro) [inline src/app/accounts/contacts/contacts.component.ts]: Done

**Comment by JuanGro** [comment]
Done

**Comment by JuanGro** [comment]
Done

**Comment by JuanGro** [comment]
DONE!

File: src/app/accounts/contacts/contacts-list.component.ts [hunk 7]
```diff
@@ -25,10 +28,14 @@ export class ContactsListComponent {
...
+  @Output() contactsListFiltered: EventEmitter<Contact[]> = new EventEmitter<Contact[]>();
```
Comment (EduardoVaca) [inline src/app/accounts/contacts/contacts-list.component.ts]: This should be removed... I believe you changed the approach

File: src/app/accounts/contacts/contacts.component.html [hunk 8]
```diff
@@ -1,19 +1,22 @@
...
 <!-- (requestShowNewContactModal) receives the request to show the new modal and initialize it -->
 <!-- (requestShowUpdateContactModal) receives the request to show the update modal -->
 <!-- (currentContact) receives the contact who was selected by the user -->
-<contacts-list [contactsList]="contactsList" [clientsList]="clientsList" (requestShowNewContactModal)="newContactModal.show()" (requestShowNewContactModal)="initializeModal()" (requestShowUpdateContactModal)="updateContactModal.show()" (currentContact)="getContactFromTable($event)"></contacts-list>
+<!-- (contactsListFiltered) receives the contacts list updated for the filter in the contacts list component -->
+<!-- (valueSearch) receives the value of the search if the user is typing something in the form -->
+<contacts-list [contactsList]="contactsList" [clientsList]="clientsList" (requestShowNewContactModal)="newContactModal.show()" (requestShowNewContactModal)="initializeModal()" (requestShowUpdateContactModal)="updateContactModal.show()" (currentContact)="getContactFromTable($event)"
+    (contactsListFiltered)="getContactsListFromTable($event)" (valueSearch)="getValueSearch($event)"></contacts-list>
```
Comment (EduardoVaca) [inline src/app/accounts/contacts/contacts.component.html]: This should be remove as well : contactsListFiltered

File: src/app/accounts/contacts/contacts.component.ts [hunk 9]
```diff
@@ -110,18 +117,66 @@ export class ContactsComponent implements OnInit {
...
+    if (this.valueSearch === '') {
+      this.contactsList = this.completeContactsList;
+    } else {
+      for (let contactFromList of this.completeContactsList) {
+        if (contactFromList.name.toLowerCase().includes(this.valueSearch.toLowerCase()) ||
+            contactFromList.last_name.toLowerCase().includes(this.valueSearch.toLowerCase()) ||
+            contactFromList.charge.toLowerCase().includes(this.valueSearch.toLowerCase()) ||
+            contactFromList.client_complete.name.toLowerCase().includes(this.valueSearch.toLowerCase()) ||
+            contactFromList.email.toLowerCase().includes(this.valueSearch.toLowerCase()) ||
+            contactFromList.alternate_email.toLowerCase().includes(this.valueSearch.toLowerCase())) {
```
Comment (EduardoVaca) [inline src/app/accounts/contacts/contacts.component.ts]: remove alternate_email from filter

File: src/app/accounts/contacts/contacts.component.html (hunk 8 above)
Comment (JuanGro) [inline src/app/accounts/contacts/contacts.component.html]: Done

File: src/app/accounts/contacts/contacts-list.component.ts (hunk 7 above)
Comment (JuanGro) [inline src/app/accounts/contacts/contacts-list.component.ts]: Done

File: src/app/accounts/contacts/contacts.component.ts (hunk 9 above)
Comment (JuanGro) [inline src/app/accounts/contacts/contacts.component.ts]: Done
